addons/website/static/src/js/content/website_root.js

```diff
@@ -33,6 +33,19 @@ export const WebsiteRoot = publicRootData.PublicRoot.extend(KeyboardNavigationMi
             autoAccessKeys: false,
             skipRenderOverlay: true,
         });
+
+        // Special case for Safari browser: padding on wrapwrap is added by the
+        // layout option (boxed, etc), but it also receives a border on top of
+        // it to simulate an addition of padding. That padding is added with
+        // the "sidebar" header template to combine both options/effects.
+        // Sadly, the border hack is not working on safari, the menu is somehow
+        // broken and its content is not visible.
+        // This class will be used in scss to instead add the border size to the
+        // padding directly on Safari when "sidebar" menu is enabled.
+        if (/^((?!chrome|android).)*safari/i.test(navigator.userAgent)) {
+            document.querySelector("#wrapwrap").classList.add("o_safari_browser");
+        }
+
```

Comment:
I would prefer this code to be in `adapt_content.js`. This file is used precisely to modify the dom when the page loads